### PR TITLE
Adaptive batching rev 3

### PR DESCRIPTION
**Pull request by harishpujari**
Preliminary pull request to see the correctness. Performance tests need to be run.

### CODE REVIEW

File: src/thrift-client/TgtInterfaceImpl.cpp [hunk 1]
```diff
@@ -760,6 +802,7 @@ std::pair<Request*, bool> StordVmdk::NewRequest(Request::Type type,
 	}
 
 	++stats_.pending_;
+	++stord_stats_.pending_;
```
Comment (prasad-joshi) [inline src/thrift-client/TgtInterfaceImpl.cpp]: this seems redundant

File: src/thrift-client/TgtInterfaceImpl.cpp [hunk 2]
```diff
@@ -768,21 +811,39 @@ std::pair<Request*, bool> StordVmdk::NewRequest(Request::Type type,
 	request->buf_sz = buf_sz;
 	request->xfer_sz = xfer_sz;
 	request->offset = offset;
+	request->batch_size = batch_size_;
 
 	auto reqp = request.get();
 	std::lock_guard<std::mutex> lock(requests_.mutex_);
-	bool schedule_now = requests_.scheduled_.empty() or
-		latency_avg_.Average() > kExpectedWanLatency;
+	//if nothing is scheduled till now, schedule this io immediately
```
Comment (prasad-joshi) [inline src/thrift-client/TgtInterfaceImpl.cpp]: please remove the commend

File: src/thrift-client/TgtInterfaceImpl.cpp (hunk 1 above)
Comment (harishpujari) [reply src/thrift-client/TgtInterfaceImpl.cpp]: stats_ is VMDK instance specific, while stord_stats_ is static across VMDK.

File: src/thrift-client/TgtInterfaceImpl.cpp (hunk 2 above)
Comment (harishpujari) [reply src/thrift-client/TgtInterfaceImpl.cpp]: done

**Pull request merged**